src/Features/CSharp/Portable/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertIfToSwitch
+{
+    internal sealed partial class CSharpConvertIfToSwitchCodeRefactoringProvider
+    {
+        private sealed class CSharpAnalyzer : Analyzer
+        {
+            public override bool SupportsSwitchExpression => true;
+            public override bool SupportsCaseGuard => true;
+            public override bool SupportsRangePattern => false;
+            public override bool SupportsTypePattern => true;
+            public override bool SupportsSourcePattern => true;
+            public override bool SupportsRelationalPattern => false;
+
+            public CSharpAnalyzer(ISyntaxFactsService syntaxFacts)
+                : base(syntaxFacts)
+            {
+            }
+
+            public override bool HasUnreachableEndPoint(IOperation operation)
+                => !operation.SemanticModel.AnalyzeControlFlow(operation.Syntax).EndPointIsReachable;
```

Comment:
interesting.  this is different from VB?

Reply:
In VB we need to use first/last overload, in C# it's not necessary but we could use it for both in the base class.

Reply:
because in VB there's no such a thing as an embedded statement, we need to analyze a list of statements, so that's the overload being used.